xds/internal/resolver/xds_resolver.go

```diff
@@ -88,26 +87,65 @@ func (b *xdsResolverBuilder) Build(target resolver.Target, cc resolver.ClientCon
 	r.logger = prefixLogger(r)
 	r.logger.Infof("Creating resolver for target: %+v", target)
 
+	// Initialize the serializer used to synchronize the following:
+	// - updates from the xDS client. This could lead to generation of new
+	//   service config if resolution is complete.
+	// - completion of an RPC to a removed cluster causing the associated ref
+	//   count to become zero, resulting in generation of new service config.
+	// - stopping of a config selector that results in generation of new service
+	//   config.
+	ctx, cancel := context.WithCancel(context.Background())
+	r.serializer = grpcsync.NewCallbackSerializer(ctx)
+	r.serializerCancel = cancel
+
+	// Initialize the xDS client.
 	newXDSClient := rinternal.NewXDSClient.(func() (xdsclient.XDSClient, func(), error))
 	if b.newXDSClient != nil {
 		newXDSClient = b.newXDSClient
 	}
-
 	client, close, err := newXDSClient()
 	if err != nil {
 		return nil, fmt.Errorf("xds: failed to create xds-client: %v", err)
 	}
 	r.xdsClient = client
 	r.xdsClientClose = close
+
+	// Determine the listener resource name and start a watcher for it.
+	template, err := r.sanityChecksOnBootstrapConfig(target, opts, r.xdsClient)
+	if err != nil {
+		return nil, err
+	}
+	endpoint := target.URL.Path
+	if endpoint == "" {
+		endpoint = target.URL.Opaque
+	}
+	endpoint = strings.TrimPrefix(endpoint, "/")
+	r.ldsResourceName = bootstrap.PopulateResourceTemplate(template, endpoint)
+	r.listenerWatcher = newListenerWatcher(r.ldsResourceName, r)
```

Comment:
Yeah let's not block this PR, but I think the e2e test cases will be useful to have. Especially if we do Mark's change with respect to moving all xDS into this resolver.

Reply:
Ah, ok. For context, we had code coverage in the listener wrapper server side, but not e2e test. So the RDS codepath hit, but the moment accept() hit, it paniced.

Reply:
The listener wrapper is a component inside the server. So, having a unit test there is not the same as the test that I'm talking about where we ensure that the output expected of the resolver is achieved, i.e service config. I agree that there is still a chance for things to fail, since we are not checking the config selector here. 

And yes, my full support for e2e style tests, and I can get started on compiling a list after this PR.